### ISSUE TITLE
Cannot get training running correctly

### ISSUE THREAD

**Issue by huangyuxiang03** [VITA-Group/LoCoCo#3]
Thanks for you amazing work!
From this repo, I assume that you have modified the original transformers package, since the usage of `past_key_values` of class `StaticCache` is not standard, and the training script directly import from transformers instead of `llama` directory. 
I can get most of the code running by manually replacing the llama code of transformers, but it seems like the modified code of `StaticCache` is not provided. This is really important for successfully running the code because I cannot infer the implementation details from the simple usage. Could you provide this part?
I think it is understandable that this work involves modification to transformers, which is really complex and hard to get everything organized in a very short time. So could you provide the modified transformers that I can directly install your version of transformers and get everything running?
Besides, could you provide a `requirements.txt` to indicate the environment prequisites? And could you also give some examples on reproducing the experimental results by using LoCoCo? (Which is the example of how to conduct inference by the given code.)
Best Wishes.

**Comment by huangyuxiang03** [comment]
Apart from the stuff above, I think it would be super nice to have more examples of applying LoCoCo to current state-of-the-art models, e.g. Llama-3.1 and mixtral.

**Comment by huangyuxiang03** [comment]
Update: I tried to fix the code and it is running now. However, I am trying to train a LoCoCo of Llama-3.1-8B-instruct, and convoluted values explode after a few steps, resulting in NaN of output. Are there any hints on fixing this?

**Comment by cairuisi** [comment]
Thank you so much for your interest in our work. I have uploaded the requirements.txt file and addressed the import errors.

Regarding your question about the NaN output, I have encountered unstable training a few times, though it is quite rare (only  a few times out of hundreds of experiments). A simple restart usually resolves the issue. However, you might want to consider applying an initialization trick for the convolutions or enabling learning rate warmup as potential solutions.

Thanks again for your interest!

Best regards,
